backend/src/crawler/udn_crawler.py

```diff
@@ -0,0 +1,138 @@
+
+class UDNCrawler(NewsCrawlerBase):
+    CHANNEL_ID = 2
+
+    def __init__(self, timeout: int = 5) -> None:
+        self.news_website_url = "https://udn.com/api/more"
+        self.news_website_news_child_urls = []
+        self.timeout = timeout
+
+    def startup(self, search_term: str) -> list[Headline]:
+        """
+        Initializes the application by fetching news headlines for a given search term across multiple pages.
+        This method is typically called at the beginning of the program when there is no data available,
+        hence it fetches headlines from the first 10 pages.
+
+        :param search_term: The term to search for in news headlines.
+        :return: A list of Headline namedtuples containing the title and URL of news articles.
+        :rtype: list[Headline]
+        """
+        return self.get_headline(search_term, page=(1, 10))
+
+    def get_headline(
+        self, search_term: str, page: int | tuple[int, int]
+    ) -> list[Headline]:
+
+        # Calculate the range of pages to fetch news from.
+        # If 'page' is a tuple, unpack it and create a range representing those pages (inclusive).
+        # If 'page' is an int, create a list containing only that single page number.
+        # page_range = range(*page) if isinstance(page, tuple) else [page]
+        headlines = []
+        if isinstance(page,tuple):
+            page_range = range(page[0],page[1]+1)
+        else:
+            page_range = [page]
+        for num in page_range:
+            headlines.extend(self._fetch_news(num,search_term))
+        return headlines
+
+    def _fetch_news(self, page: int, search_term: str) -> list[Headline]:
+        response = self._perform_request(self.news_website_url,
+                                         self._create_search_params(page, search_term, "searchword"))
+        return self._parse_headlines(response)
+
+    def _create_search_params(self, page: int, search_term: str, type: str = "searchword") -> dict:
+        return {
+            "page": page,
+            "id": f"search:{quote(search_term)}",
+            "channelId": self.CHANNEL_ID,
+            "type": type,
+        }
+
+    def _perform_request(self, url: str | None = None, params: dict | None = None) -> Response:
+        return get(url, params)
+            
+
+    @staticmethod
+    def _parse_headlines(response: Response) -> list[Headline]:
+        news_list = response.json()["lists"]
+        processed_headlines_list = []
+        for news in news_list:
+            headline = Headline(title=news["title"], url=news["titleLink"])
+            processed_headlines_list.append(headline)
+        return processed_headlines_list
+
+    def parse(self, url: str) -> News:
+        response = self._perform_request(url)
+        if not self._is_valid_url(url):
+            raise DomainMismatchException(url)
+        return self._extract_news(BeautifulSoup(response.text, "html.parser"), url)
```

Comment:
This function won't work since no parameters of the search request are provided to `self._perform_request`. Additionally, as mentioned above, this method may be intended to be called only by `validate_and_parse`. If that's the case, it might be better to remove the code, as the received URL has already been verified..